src/nodes/display/NormalMapNode.js

```diff
@@ -57,14 +58,28 @@ class NormalMapNode extends TempNode {
 		 */
 		this.normalMapType = TangentSpaceNormalMap;
 
+		/**
+		 * Controls whether the Z component of the normal map is reconstructed projecting it to the hemisphere.
+		 *
+		 * @type {bool}
+		 * @default false
+		 */
+		this.projectZ = false;
+
 	}
 
 	setup( { material } ) {
 
-		const { normalMapType, scaleNode } = this;
+		const { normalMapType, scaleNode, projectZ } = this;
 
 		let normalMap = this.node.mul( 2.0 ).sub( 1.0 );
 
+		if ( projectZ ) {
+
+			normalMap = vec3( normalMap.xy, sqrt( saturate( float( 1.0 ).sub( dot( normalMap.xy, normalMap.xy ) ) ) ) );
```

Comment:
Do you mind moving this code into`src/nodes/utils/Packing.js` into a new function? It might be handy elsewhere.

```js
/**
 * Unpacks RG-packed normals by reconstructing their Z component by projecting it to the hemisphere.
 *
 * @tsl
 * @function
 * @param {Node<vec3>} node - The color to unpack.
 * @return {Node<vec3>} The direction.
 */
export const unpackRGNormal = ( normal ) => vec3( normal.xy, sqrt( saturate( float( 1.0 ).sub( dot( normal.xy, normal.xy ) ) ) ) );
```
We can then use it in `NormalMapNode` like so:
```js
if ( useRGPacking === true ) {

    normalMap = unpackRGNormal( normalMap );

}